client.go

```diff
@@ -376,10 +388,20 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 			break
 		}
 	}
-
 	resp.Body = ioutil.NopCloser(bytes.NewReader([]byte{}))
 	conn.subprotocol = resp.Header.Get("Sec-Websocket-Protocol")
 
+	// Need to handle the case of there being remaining buffered bytes in the limited reader
+	if maxHeaderSizeIsSet {
```

Comment:
What is the symptom if this is not done? As in, if you don't handle this case, are we leaking memory or dropping data?

Reply:
It's worse than that. If we don't take out retrieve any already buffered data from the bufferedreader we would have misaligned websocket frames and the connection would be effectively dead.